### PR TITLE
add autofix for `PYI030`

### LINKED ISSUE #7854: Request: PYI030 autofix
I am using `ruff==0.0.292`, and am turning on `PYI`.  I am wondering, why doesn't `PYI030` have an autofix?

It seems possible to at least support partial autofixes here, any chance we can add this capability?

### PR DESCRIPTION
**Pull request by diceroll123**
## Summary

Adds autofix to `PYI030`

Closes #7854. 

Unsure if the cloning method I chose is the best solution here, feel free to suggest alternatives!

## Test Plan

`cargo test` as well as manually

### CODE REVIEW

File: crates/ruff_linter/src/rules/flake8_pyi/rules/unnecessary_literal_union.rs [hunk 1]
```diff
@@ -59,16 +63,23 @@ pub(crate) fn unnecessary_literal_union<'a>(checker: &mut Checker, expr: &'a Exp
...
-                members: literal_exprs
-                    .into_iter()
-                    .map(|expr| checker.locator().slice(expr.as_ref()).to_string())
-                    .collect(),
+                members: literal_members.clone(),
             },
             expr.range(),
         );
 
+        diagnostic.set_fix(Fix::safe_edit(Edit::range_replacement(
```
Comment (charliermarsh) [inline crates/ruff_linter/src/rules/flake8_pyi/rules/unnecessary_literal_union.rs:78]: Do you mind gating this in an `if checker.patch(diagnostic.kind.rule()) { ... }` block? We allow users to turn autofixes on and off, and that's enforced through this pattern.

File: crates/ruff_linter/src/rules/flake8_pyi/rules/unnecessary_literal_union.rs [hunk 2]
```diff
@@ -59,16 +63,23 @@ pub(crate) fn unnecessary_literal_union<'a>(checker: &mut Checker, expr: &'a Exp
...
+            format!("Literal[{}]", literal_members.clone().join(", ")),
```
Comment (charliermarsh) [inline crates/ruff_linter/src/rules/flake8_pyi/rules/unnecessary_literal_union.rs:79]: Can this `.clone()` be removed? We don't use `literal_memebrs` again, so it should be fine to consume it here.

File: crates/ruff_linter/src/rules/flake8_pyi/rules/unnecessary_literal_union.rs [hunk 3]
```diff
@@ -31,14 +31,18 @@ pub struct UnnecessaryLiteralUnion {
...
     #[derive_message_formats]
     fn message(&self) -> String {
         format!(
             "Multiple literal members in a union. Use a single literal, e.g. `Literal[{}]`",
             self.members.join(", ")
         )
     }
+
+    fn fix_title(&self) -> String {
+        format!("Replace Literal unions with a single Literal.")
```
Comment (charliermarsh) [inline crates/ruff_linter/src/rules/flake8_pyi/rules/unnecessary_literal_union.rs:44]: Nit: "Replace with a single `Literal`" would better match our style for these messages.

**Review by charliermarsh — APPROVED** [review]
Thanks!

**Comment by jamesbraza** [comment]
Thank you @diceroll123 and @charliermarsh!! 🥳

File: crates/ruff_linter/src/rules/flake8_pyi/snapshots/ruff_linter__rules__flake8_pyi__tests__PYI030_PYI030.py.snap [hunk 4]
```diff
@@ -63,34 +129,78 @@ PYI030.py:25:9: PYI030 Multiple literal members in a union. Use a single literal
...
 33 | # Should handle user parentheses when fixing.
    |
+   = help: Replace with a single `Literal`
+
+ℹ Fix
+28 28 | field7 = Literal[1] | Literal[2]  # Error
+29 29 | 
+30 30 | # Should emit for parenthesized unions.
+31    |-field8: Literal[1] | (Literal[2] | str)  # Error
+   31 |+field8: Literal[1, 2]  # Error
```
Comment (T-256) [inline crates/ruff_linter/src/rules/flake8_pyi/snapshots/ruff_linter__rules__flake8_pyi__tests__PYI030_PYI030.py.snap:179]: Here expected `Literal[1, 2] | str` is the correct fix (?)

**Pull request merged**